core/src/main/scala/filodb.core/memstore/TimeSeriesShard.scala

```diff
@@ -923,31 +936,72 @@ class TimeSeriesShard(val dataset: Dataset,
   private[filodb] def getOrAddPartition(recordBase: Any, recordOff: Long, group: Int, ingestOffset: Long) = {
     var part = partSet.getWithIngestBR(recordBase, recordOff)
     if (part == null) {
-      part = addPartition(recordBase, recordOff, group, ingestOffset)
+      part = addPartition(recordBase, recordOff, group)
     }
     part
   }
   // scalastyle:on
 
-  private def addPartition(recordBase: Any, recordOff: Long, group: Int, ingestOffset: Long) = {
+  private def lookupPreviouslyAssignedPartId(partKeyBase: Array[Byte], partKeyOffset: Long): Option[Int] = {
+    if (evictedPartKeys.mightContain(PartKey(partKeyBase, partKeyOffset))) {
+      val filters = dataset.partKeySchema.toStringPairs(partKeyBase, partKeyOffset)
+        .map { pair => ColumnFilter(pair._1, Filter.Equals(pair._2)) }
+      val matches = partKeyIndex.partIdsFromFilters2(filters, 0, Long.MaxValue)
+      matches.result.cardinality() match {
+        case 1 =>           val partId = matches.result.intIterator().next()
+                            logger.debug(s"There is already a partId $partId assigned for " +
+                              s"${dataset.partKeySchema.stringify(partKeyBase, partKeyOffset)}")
+                            Some(partId)
+        case 0 =>           shardStats.evictedPartKeyBloomFilterFalsePositives.increment()
+                            None
+        case c if c > 1 =>  val iter = matches.result.intIterator()
```

Comment:
Can we have a counter for the # of times that the match returns more than 1?   This is an especially expensive path so we want to see how often it's invoked.

Reply:
Basically, what we want to see:
- When we need to add partitions, how often do we need to actually search Lucene (hopefully bloom filter hit rate is low)
- When we search Lucene, how often we need to compare multiple values

Reply:
will add.